src/cai-log.ads

```diff
@@ -1,23 +1,50 @@
+--
+--  @summary Log interface declarations
+--  @author  Johannes Kliemann
+--  @date    2019-04-10
+--
+--  Copyright (C) 2019 Componolit GmbH
+--
+--  This file is part of ada-interface, which is distributed under the terms of the
+--  GNU Affero General Public License version 3.
+--
 
 private with Cai.Internal.Log;
 
 package Cai.Log with
    SPARK_Mode
 is
 
+   --  Catch all mod type, only used for image functions
```

Comment:
What is that used for - could it be private?

Reply:
It is public to allow casting. Since I cannot create a catch-all image function for all possible mod types you can cast your custom mod type into this one and use it with the image function. If it was private it cannot be casted.

Reply:
Solution: Create a generic function with a discrete generic formal type e.g. Data_Type is (<>).

Reply:
We'll solve this later, cf. #52